daffodil-tdml-processor/src/main/scala/org/apache/daffodil/processor/tdml/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -113,6 +113,7 @@ final class TDMLDFDLProcessorFactory private (
       val diags = p.getDiagnostics
       Left(diags)
     } else {
+      val diags = p.getDiagnostics
```

Comment:
`val diags = p.getDiagnostics` is now duplicated as the first thing in each if-block branch. Can we move this outside of the if block to remove the duplication?